pkg/controller/config/config_controller_test.go

```diff
@@ -0,0 +1,206 @@
+		)
+
+		// GIVEN
+		os.Setenv("IMAGE_TRIGGERS_"+strings.ToUpper(container), image)
+		os.Setenv("IMAGE_TRIGGERS_"+strings.ToUpper(arg), argImage)
+		config := configFor(configName, namespace)
+		cl := feedConfigMock(config)
+		addons, err := mfFor("addons", cl)
+		assetNoEror(err, "failed to create manifestival for triggers addons;", t)
+		req := requestFor(configName, namespace)
+		r := ReconcileConfig{scheme: scheme.Scheme, client: cl, addons: addons}
+
+		// WHEN
+		_, err = r.applyAddons(req, config)
+
+		// THEN
+		assetNoEror(err, "failed to reconcile for applyPipeline;", t)
+		assertContainerHasImage(deployment, container, image, r.client, t)
+		assertContainerArgHasImage(deployment, arg, argImage, r.client, t)
+	})
+}
+
+func configFor(name string, namespace string) *op.Config {
+	return &op.Config{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: op.ConfigSpec{
+			TargetNamespace: namespace,
+		},
+	}
+}
+
+func requestFor(configName string, namespace string) reconcile.Request {
+	return reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      configName,
+			Namespace: namespace,
+		},
+	}
+}
+
+func assetNoEror(err error, msg string, t *testing.T) {
+	t.Helper()
+
+	if err != nil {
+		t.Errorf("%s expected no error, %v", msg, err)
+	}
+}
+
+func assertContainerHasImage(deploy string, container string, image string, cl client.Client, t *testing.T) {
+	t.Helper()
+
+	containers := deploymentFor(deploy, cl, t)
+
+	for _, c := range containers {
+		if c.Name != container {
+			continue
+		}
+
+		if c.Image != image {
+			t.Fatalf("assertion failed; expected container %s but got %s", image, c.Image)
+		}
+	}
+}
+
+func assertContainerArgHasImage(deploy string, arg string, image string, cl client.Client, t *testing.T) {
+	t.Helper()
+
+	containers := deploymentFor(deploy, cl, t)
+
+	for _, container := range containers {
+		if container.Args == nil || len(container.Args) == 0 {
+			continue
+		}
+
+		for a, argument := range container.Args {
+			if argument != arg {
+				continue
+			}
+
+			if container.Args[a+1] != image {
+				t.Fatalf("assertion failed; expected arg image %s but got %s", image, container.Args[a+1])
+			}
+		}
+	}
+}
+
+func deploymentFor(name string, cl client.Client, t *testing.T) []v1.Container {
+	dep := &appsv1.Deployment{}
+	namespacedName := types.NamespacedName{
+		Name:      name,
+		Namespace: "openshift-pipelines",
+	}
+
+	err := cl.Get(context.TODO(), namespacedName, dep)
+	if err != nil {
+		t.Fatalf("assertion failed; get deployment: (%v)", err)
+	}
+
+	containers := dep.Spec.Template.Spec.Containers
+
+	if containers == nil && len(containers) == 0 {
+		t.Fatalf("assertion failed; deployment not containers")
+	}
+
+	return containers
+}
+
+func mfFor(resource string, cl client.Client) (mf.Manifest, error) {
+	_, filename, _, _ := rt.Caller(0)
+	root := path.Join(path.Dir(filename), "../../..")
```

Comment:
should we keep some test data in ` pkg/controller/config/` instead of using the `deploy/resources/...`